packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/v5/CodeMirror.tsx

```diff
@@ -350,6 +351,17 @@ function CodeMirror(props: NoteBodyEditorProps, ref: ForwardedRef<NoteBodyEditor
 		});
 	};
 
+	// Since the editor is not yet rendered when no notes are opened, this is necessary
+	useAsyncEffect(async event => {
+		const n = await Note.load(props.noteId);
+		if (event.cancelled) return;
+		const focusSettingName = n.is_todo ? 'newTodoFocus' : 'newNoteFocus';
+
+		if (Setting.value(focusSettingName) === 'body') {
+			if (editorRef.current) editorRef.current.focus();
+		}
+	}, [editorRef.current, props.noteId]);
```

Comment:
The focus should be handled where you removed the code from. Now we have `handleAutoFocus` which may or may not handle auto focus (I think it does nothing now), and we have code here that handle auto-focus but only for CodeMirror 5.

Reply:
The problem that I realized while debugging this is that if I were to handle it in handleAutoFocus, the execCommand returns false and never fires handleAutoFocus due to a race condition as the CodeMirror editor likely hasn't been rendered yet according to the fact that editorRef is undefined, and stepping through the devtools debugger does not reproduce the issue. Awaiting the focus command was the first thing I tried to see if that would fix it but to no avail. 

I'll experiment a bit more with keeping it in the handleAutoFocus, but in the case of moving it to an effect, would you prefer if I were to move all the autoFocus handling into an effect and apply the same to all the CodeMirror and other editor versions?

Reply:
I don't really know the technical details but, especially when it comes to focus-related code, it's better to have it all in one place so ideally in the existing `handleAutoFocus`. Or you can move it elsewhere but there should only be one. Maybe you need to look at the state variable and ensure that everything is loaded correctly before running the focus handling code

Reply:
You gave me a good idea and I believe I was overthinking it by quite a bit. I've changed it to using a timeout recursive loop to wait a little bit for the state variable to load properly and it should handle it a lot more neatly now. In my own testing, it only reruns once if the editor is not rendered at first (i.e. a user has not opened a note or todo) so this should have extremely minimal impact on performance.